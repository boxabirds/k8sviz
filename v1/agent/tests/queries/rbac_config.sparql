# Query: Create RBAC configuration for a service account
# This query retrieves RBAC resources and relationships

PREFIX k8s: <http://kubernetes.io/ontology#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?resource ?property ?value ?comment ?constraint
WHERE {
  # Get RBAC resources
  {
    ?rbacResource rdfs:subClassOf k8s:APIObject ;
                  rdfs:label ?label ;
                  rdfs:comment ?comment .
    FILTER(CONTAINS(STR(?rbacResource), "Role") || 
           CONTAINS(STR(?rbacResource), "ServiceAccount"))
  }
  
  # Get RBAC properties
  UNION
  {
    ?property rdfs:label ?propertyLabel ;
              rdfs:comment ?propertyComment .
    FILTER(CONTAINS(STR(?property), "role") || 
           CONTAINS(STR(?property), "permission") ||
           CONTAINS(STR(?property), "subject"))
  }
  
  # Get RBAC verbs
  UNION
  {
    ?verb a k8s:RBACVerb ;
          rdfs:label ?verbLabel .
  }
  
  # Get RBAC constraints
  UNION
  {
    ?constraint rdfs:label ?constraintLabel ;
                rdfs:comment ?constraintComment .
    FILTER(CONTAINS(LCASE(STR(?constraintComment)), "rbac") || 
           CONTAINS(LCASE(STR(?constraintComment)), "permission"))
  }
  
  # Get RBAC best practices
  UNION
  {
    ?practice a k8s:BestPractice ;
              rdfs:comment ?practiceComment .
    FILTER(CONTAINS(LCASE(STR(?practiceComment)), "rbac") || 
           CONTAINS(LCASE(STR(?practiceComment)), "privilege"))
  }
}