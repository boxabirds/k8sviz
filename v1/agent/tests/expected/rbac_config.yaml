---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: app-service-account
  namespace: default
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: app-role
  namespace: default
rules:
- apiGroups: [""]
  resources: ["pods", "services"]
  verbs: ["get", "list", "watch"]
- apiGroups: [""]
  resources: ["configmaps"]
  verbs: ["get", "list", "watch", "create", "update", "patch"]
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get", "list"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: app-role-binding
  namespace: default
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: app-role
subjects:
- kind: ServiceAccount
  name: app-service-account
  namespace: default